src/main/kotlin/com/refinedmods/refinedstorage/data/sync/C2SSyncedData.kt

```diff
@@ -0,0 +1,41 @@
+package com.refinedmods.refinedstorage.data.sync
+
+import io.netty.buffer.PooledByteBufAllocator
+import net.fabricmc.fabric.api.network.ClientSidePacketRegistry
+import net.fabricmc.fabric.api.network.PacketConsumer
+import net.fabricmc.fabric.api.network.PacketContext
+import net.fabricmc.fabric.api.network.ServerSidePacketRegistry
+import net.minecraft.entity.data.TrackedDataHandler
+import net.minecraft.entity.player.PlayerEntity
+import net.minecraft.network.PacketByteBuf
+import net.minecraft.util.Identifier
+
+/**
+ * Watches underlying data for changes from the client, and
+ * sync's changes to the server. The client is the source of truth.
+ */
+class C2SSyncedData<T: SimpleObservable>(
+    identifier: Identifier,
+    isClient: Boolean,
+    internalData: T,
+    serializer: TrackedDataHandler<T>,
+    player: PlayerEntity,
+    onChanged: ((T)->Unit)? = null
+) : BiSyncedData<T>(identifier, isClient, internalData, serializer, player, onChanged) {
+
+    override fun send() {
+        val byteBuffer = byteBuffers.buffer()
+        try {
+            val buf = PacketByteBuf(byteBuffer)
+            this.serializer.write(buf, internalData)
+            when(isClient) {
+                true -> ClientSidePacketRegistry.INSTANCE.sendToServer(identifier, buf)
+                false -> {} // NO-OP on server
```

Comment:
Maybe check this first, before the rest of the logic?

Reply:
It's inside the is Client check. In a C2S setup (client to server) only the client tracks changes and notifies the server. The server only receives updates and thus has a noop for their send.

Reply:
I mean convert
```kotlin
override fun send() {
        val byteBuffer = byteBuffers.buffer()
        try {
            val buf = PacketByteBuf(byteBuffer)
            this.serializer.write(buf, internalData)
            when(isClient) {
                true -> ClientSidePacketRegistry.INSTANCE.sendToServer(identifier, buf)
                false -> {} // NO-OP on server
            }
        } finally { byteBuffer.release() }
    }
```
into
```kotlin
    override fun send() {
        if(!isClient) return
        val byteBuffer = byteBuffers.buffer()
        try {
            val buf = PacketByteBuf(byteBuffer)
            this.serializer.write(buf, internalData)
            ClientSidePacketRegistry.INSTANCE.sendToServer(identifier, buf)
        } finally { byteBuffer.release() }
    }
```

Reply:
Yeah, I was think about that while typing the first comment; no point in writing out to a buffer only to discard it. I agree.